core/src/epicli/data/common/ansible/playbooks/roles/sshtunnel/files/tunnel.sh

```diff
@@ -1,12 +1,48 @@
 #!/bin/bash
 
-set -o errexit
+set -o errexit -o nounset -o pipefail
+
+export AUTOSSH_PIDFILE=/tmp/autossh-kubectl-tunnel.pid
+
+is_port_available() {
+  local port=$1
+
+  # nc command is much faster than lsof and doesn't require sudo to take into account ports used by other users
+  nc -z 127.0.0.1 "$port" && return 1 || return 0 # 0 if port is available
+}
 
 if [ "$1" = "--kill" ]; then
-  ps -C ssh,autossh -o pid,cmd |
-  grep -E "@${2}($|[[:blank:]])" |
-  awk '{print $1}' |
-  xargs -r kill
-else
+  LOCAL_PORT=$2
+
+  # kill autossh (ssh is child process)
+  if test -f $AUTOSSH_PIDFILE; then
```

Comment:
How about (for consistency reasons)? :thinking: 
```suggestion
  if [ -f "$AUTOSSH_PIDFILE" ]; then
```